### ISSUE TITLE
Progress indicator tends to be useless

### ISSUE THREAD

**Issue by MichaelAtOz** [openscad/openscad#810]
The counter stays at 0/1000 until the render is finished.

TODO:
- [ ] Don't use CGAL's Nef3_nary_union - rather do this ourselves and provide a progress/cancel option.
##

**Comment by thehans** [comment]
Is any of this still an issue?  The first two examples from @MichaelAtOz seem to progress reasonably for me. 

example006 does seem to jump up to ~680 fairly quickly, then spend a bit more time towards the end of the bar, but it does at least update progress pretty regularly.  Definitely not staying at 0/1000 the whole time for me.

I also glanced at #803 looking for 'Large' but didn't immediately see what script @MichaelAtOz was referring to.

**Comment by thehans** [comment]
OK, I just realized github had a lot of collapsed comments on that page, I think this is the 'Large' example:
```
t = [
    [0, "0123456789"],
    [1, "ab   cdefghijklm"],
//  [1, "a        bcdefghijklmo"],
    [2, " nopqrstuvwxyz"],
    [3, "ABCDEFGHIJKLM"],
    [4, "NOP   QRSTUVWXYZ"],
    //[5, "!@#$%^&*(;)],"],
    //[6, "/\"|'\\~`_+-∰"],
    [7, [127,126,125] ],
];

for (x = [0:300:15000])
for (a = t) {
//echo(x=x,a=a);
    translate([x/2,-16 * a[0] ])
        text(a[1], font = "liberation sans");
}
```
F5 finishes pretty much immediately for me, so 0 seconds reported seems more or less accurate.  No time for progress bar to appear.
F6 takes about 6 seconds, just counting in my head, but does still show 0 seconds in console output, so that does seem to still be an issue.  Progress bar does not appear for me at all during that time.

**Comment by Jimw338** [comment]
Just noticed this - annoying as I don't know if it's just not done, or has failed somehow (infinite loop, crash, etc).  Would it be possible to have it go through the render once to count the number of "calls" to various routines - count them but not actually do any calculations.  And then go through it again, which would let the progress bar (sort of) work.  It would work as far as indicating "how many operations of the total have been completed", but not necessarily with reference to the time they take.

I haven't dug into the code *at all*, so this make no sense at all.

**Comment by t-paul** [comment]
@Jimw338 Nope, the problem is not that the number of operations is unknown, the problem is that the last operation could potentially take orders of magnitude more time and there's no easy estimation of that.
